### PR TITLE
[docs] Fix encryption examples

### PR DESCRIPTION
**Pull request by trexinc**
### Motivation

Examples for encryption were incorrect, they were a mix of Java and C++ and could not be used as is.

### Modifications

Update the examples code and some texts

### Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: site2/docs/security-encryption.md [hunk 1]
```diff
@@ -160,16 +165,15 @@ When producers want to encrypt the messages with multiple keys, producers add al
...
-  * If consumption fails due to decryption failure or missing keys in consumer, application has the option to consume the encrypted message or discard it. Call conf.setCryptoFailureAction(ConsumerCryptoFailureAction) to control the consumer behavior. The default behavior is to fail the request. Application is never able to decrypt the messages if the private key is permanently lost.
+  * Producer action fails indicating the cause of the failure. Application has the option to proceed with sending unencrypted message in such cases. Call PulsarClient.newProducer().cryptoFailureAction(ProducerCryptoFailureAction.SEND) to control the producer behavior. The default behavior is to fail the request.
+  * If consumption fails due to decryption failure or missing keys in consumer, application has the option to consume the encrypted message or discard it. Call PulsarClient.newConsumer().cryptoFailureAction(ConsumerCryptoFailureAction.DISCARD) to control the consumer behavior. The default behavior is to fail the request. Application is never able to decrypt the messages if the private key is permanently lost.
```
Comment (jiazhai) [inline site2/docs/security-encryption.md:176]: In above 2 lines `SEND` and `DISCARD` should be removed?

**Comment by jiazhai** [comment]
Thanks for the work. left a comments

File: site2/docs/security-encryption.md (hunk 1 above)
Comment (trexinc) [reply site2/docs/security-encryption.md]: Done

**Comment by sijie** [comment]
@jiazhai can you review it again?

File: site2/docs/security-encryption.md [hunk 2]
```diff
@@ -144,11 +148,12 @@ for (int i = 0; i < 10; i++) {
 
 // Acknowledge the consumption of all messages at once
 consumer.acknowledgeCumulative(msg);
+consumer.close();
 pulsarClient.close();
 ```
 
 ## Key rotation
-Pulsar generates new AES data key every 4 hours or after publishing a certain number of messages. A producer fetches the asymmetric public key every 4 hours by calling CryptoKeyReader::getPublicKey() to retrieve the latest version.
+Pulsar generates new AES data key every 4 hours or after publishing a certain number of messages. A producer fetches the asymmetric public key every 4 hours by calling CryptoKeyReader.getPublicKey() to retrieve the latest version.
```
Comment (Anonymitaet) [inline site2/docs/security-encryption.md:156]:
```suggestion
Pulsar generates a new AES data key every 4 hours or after publishing a certain number of messages. A producer fetches the asymmetric public key every 4 hours by calling CryptoKeyReader.getPublicKey() to retrieve the latest version.
```

File: site2/docs/security-encryption.md [hunk 3]
```diff
@@ -78,18 +78,20 @@ class RawFileKeyReader implements CryptoKeyReader {
...
 
-ProducerConfiguration prodConf = new ProducerConfiguration();
-prodConf.setCryptoKeyReader(new RawFileKeyReader("test_ecdsa_pubkey.pem", "test_ecdsa_privkey.pem"));
-prodConf.addEncryptionKey("myappkey");
+PulsarClient pulsarClient = PulsarClient.builder().serviceUrl("pulsar://localhost:6650").build();
 
-Producer producer = pulsarClient.createProducer("persistent://my-tenant/my-ns/my-topic", prodConf);
+Producer producer = pulsarClient.newProducer()
+                .topic("persistent://my-tenant/my-ns/my-topic")
+                .addEncryptionKey("myappkey")
```
Comment (kjaggann) [inline site2/docs/security-encryption.md:86]: what will be the myappkey

Comment (trexinc) [reply site2/docs/security-encryption.md:86]: Its just a “tag” for the key. So the consumer knows with which key to decrypt. As basically every producer can use a different encryption key

Comment (kjaggann) [reply site2/docs/security-encryption.md:86]: I used secured  AES key generated from this method
	Key key;
		SecureRandom rand = new SecureRandom();
		KeyGenerator generator = KeyGenerator.getInstance("AES");
		generator.init(256, rand);
		key = generator.generateKey();
and  created a producer
Producer<byte[]> producer = client.newProducer()
					.topic("persistent://my-tenant/my-namespace/my-topic-1")
					.cryptoKeyReader(new RawFileKeyReader("src/main/resources/test_ecdsa_pubkey.pem", "src/main/resources/test_ecdsa_privkey.pem"))
					.addEncryptionKey("key")
					.enableBatching(false)
					.create();

But i got this exception
org.apache.pulsar.client.api.PulsarClientException: java.util.concurrent.ExecutionException: org.apache.pulsar.client.api.PulsarClientException: java.lang.NoSuchMethodError: 'org.apache.pulsar.shade.io.netty.buffer.ByteBuf org.apache.pulsar.client.api.MessageCrypto.encrypt(java.util.Set, org.apache.pulsar.client.api.CryptoKeyReader, java.util.function.Supplier, org.apache.pulsar.shade.io.netty.buffer.ByteBuf)'

------------------------------------------------------------------------------------------------------------------
I followed the document

Comment (trexinc) [reply site2/docs/security-encryption.md:86]: Yep, this seems to be completely broken in 2.5.1. Works fine in 2.5.0.

Comment (trexinc) [reply site2/docs/security-encryption.md:86]: Well, it works fine if you add this dependency:
compile group: 'org.apache.pulsar', name: 'pulsar-client-messagecrypto-bc', version: '2.5.1'

Don't know if this a bug or this is how it is supposed to be now with the whole addition of bcfips support.

Comment (kjaggann) [reply site2/docs/security-encryption.md:86]: I have already added that dependency 
however i have raised a ticket for this 

ISSUE-6834: Apache pulsar Encryption failed using ecdsa

**Pull request merged**